src/Tools/dotnet-watch/BrowserRefresh/src/BrowserRefreshMiddleware.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Watch.BrowserRefresh
+{
+    public class BrowserRefreshMiddleware
+    {
+        private static readonly MediaTypeHeaderValue _textHtmlMediaType = new MediaTypeHeaderValue("text/html");
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+
+        public BrowserRefreshMiddleware(RequestDelegate next, ILogger<BrowserRefreshMiddleware> logger) =>
+            (_next, _logger) = (next, logger);
+
+        public async Task InvokeAsync(HttpContext context)
```

Comment:
How do I disable this?

Reply:
There are environment switches for dotnet-watch to turn this off: 

`DOTNET_WATCH_SUPPRESS_BROWSER_REFRESH=0`. I should update the documentation for this.